src/modules/joke.ts

```diff
@@ -0,0 +1,64 @@
+/**
+ * @file
+ * Modules:
+ *  - {@link joke}
+ * Description:
+ *  Uses the joke api provided by arc to parse a response and send it as a discord embed
+ */
+
+import * as util from '../core/util.js';
+import {botConfig} from '../core/config.js';
+
+// Config manager for nsfw detection
+const jokeConfig = botConfig.modules.joke;
+let JOKE_API_URL = '';
```

Comment:
Typically `SCREAMING_SNAKE_CASE`  is used for global constants, but you have something like `BASE_JOKE_API_URL`, then have a separate variable like `blacklistFlags` that's an array of flags that get joined together, eg:
```typescript
let jokeApiUrl = 'https://v2.jokeapi.dev/joke/Any';
const blacklistFlags = ['religious', 'political', 'racist', 'sexist', 'explicit'];
if (jokeConfig.nsfw) {
  blacklistFlags.push('nsfw');
}
jokeUrl += `?blacklistFlags=${blacklistFlags.join(',')}&type=single'}`;
```
This is one way you could implement it, but please don't use `SCREAMING_SNAKE_CASE` for variables that are mutated.